src/lib/components/ui/input/color-input.svelte

```diff
@@ -3,35 +3,49 @@
 	import Button from '../button.svelte'
 	import Input, { type Props } from './input.svelte'
 	import type { HTMLInputAttributes } from 'svelte/elements'
-	let { dimension, disabled, ...restProps }: Props & HTMLInputAttributes = $props()
+	let { dimension, disabled, value, ...restProps }: Props & HTMLInputAttributes = $props()
 	let size: 16 | 24 | 32 = $derived(dimension === 'large' ? 32 : dimension === 'small' ? 16 : 24)
-	let color = $state()
+
+	let colorInput: HTMLInputElement | undefined = $state(undefined)
+	let open = $state(false)
 </script>
 
 {#snippet buttons()}
-	<Button {dimension} {disabled} variant="secondary" style="padding: 0;">
-		<label>
-			<div class="palette-icon">
-				<ColorPalette {size} />
-			</div>
-			<input type="color" bind:value={color} />
-		</label>
+	<Button
+		class="color-button"
+		{dimension}
+		{disabled}
+		variant="secondary"
+		style="padding:0;"
+		onclick={() => {
+			if (!open) {
+				colorInput?.click()
+				open = true
+			} else {
+				open = false
+			}
+		}}
+		onblur={() => {
+			open = false
+		}}
+		onkeypress={(e:KeyboardEvent)=>{
+			if(e.key === 'Enter'){
+                open = false
+            }
+		}}
+	>
```

Comment:
I also want to close. It should also work with spacebar. Probably we don't need this `onkeypress` listener, because in HTML pressing spacebar or enter on focused element should trigger the `onclick` event.